pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -244,22 +282,33 @@ def cluster_environment(self) -> ClusterEnvironment:
     def on_cpu(self) -> bool:
         return self._device_type == DeviceType.CPU
 
+    @property
+    def use_cpu(self) -> bool:
+        return self.on_cpu
+
     @property
     def on_tpu(self) -> bool:
         return self.tpu_cores is not None
 
+    @property
+    def use_tpu(self) -> bool:
+        return self._accelerator_type == DeviceType.TPU and self.on_tpu
+
```

Comment:
struggling to understand from code right now, could you give me a hint what is the difference between use_x and on_x?

If someone subclasses TPUAccelerator let's say, and passes in `accelerator=MyTPUAccelerator()`, can we expect that `on_tpu` is set True?

I have worked on this connector myself but have forgotten the details xD Apologies

Reply:
> struggling to understand from code right now, could you give me a hint what is the difference between use_x and on_x?

With the existing design, when the User passes `tpu_cores=8`, it's expected to and will run on TPUs. But post this PR, the user could do `(acclerator="cpu", tpu_cores=8)` to run it on CPU.
That's the difference with having `on_x` and `use_x`.

But, more I think about it, I should rename it as
```python
`on_x` ====> `has_x` (Suggests we have the following device X to work with)
`use_x` ===> `on_x` (As it would mean Training will be performed on X, consistent with other `on_x` properties)
```

> If someone subclasses TPUAccelerator let's say, and passes in accelerator=MyTPUAccelerator(), can we expect that on_tpu is set True?

For sure, if the User would pass `tpu_cores`, which is expected.

Reply:
Decided to go `has_x` and `use_x`, much cleaner and easy to understand ^^

Reply:
that sounds great.